lib/ansible/modules/cloud/azure/azure_rm_sqldatabase.py

```diff
@@ -0,0 +1,576 @@
+                type='datetime'
+            ),
+            recovery_services_recovery_point_resource_id=dict(
+                type='str'
+            ),
+            edition=dict(
+                type='str',
+                choices=['web',
+                         'business',
+                         'basic',
+                         'standard',
+                         'premium',
+                         'free',
+                         'stretch',
+                         'data_warehouse',
+                         'system',
+                         'system2']
+            ),
+            max_size_bytes=dict(
+                type='str'
+            ),
+            requested_service_objective_id=dict(
+                type='str'
+            ),
+            requested_service_objective_name=dict(
+                type='str',
+                choices=['basic',
+                         's0',
+                         's1',
+                         's2',
+                         's3',
+                         'p1',
+                         'p2',
+                         'p3',
+                         'p4',
+                         'p6',
+                         'p11',
+                         'p15',
+                         'system',
+                         'system2',
+                         'elastic_pool']
+            ),
+            elastic_pool_name=dict(
+                type='str'
+            ),
+            read_scale=dict(
+                type='str',
+                choices=['enabled',
+                         'disabled']
+            ),
+            sample_name=dict(
+                type='str',
+                choices=['adventure_works_lt']
+            ),
+            zone_redundant=dict(
+                type='str'
+            ),
+            state=dict(
+                type='str',
+                default='present',
+                choices=['present', 'absent']
+            )
+        )
+
+        self.resource_group = None
+        self.server_name = None
+        self.name = None
+        self.parameters = dict()
+
+        self.results = dict(changed=False)
+        self.mgmt_client = None
+        self.state = None
+        self.to_do = Actions.NoAction
+
+        super(AzureRMDatabases, self).__init__(derived_arg_spec=self.module_arg_spec,
+                                               supports_check_mode=True,
+                                               supports_tags=False)
+
+    def exec_module(self, **kwargs):
+        """Main module execution method"""
+
+        for key in list(self.module_arg_spec.keys()):
+            if hasattr(self, key):
+                setattr(self, key, kwargs[key])
+            elif kwargs[key] is not None:
+                if key == "location":
+                    self.parameters["location"] = kwargs[key]
+                elif key == "collation":
+                    self.parameters["collation"] = kwargs[key]
+                elif key == "create_mode":
```

Comment:
Instead of putting a manual case for each option, use the camel/snake converted in the ec2 module_util to convert it for you https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/ec2.py#L415-L443.

You can use it to convert values from `non_readable_secondary` to `NonReadableSecondary` and vice versa. We should potentially look at moving it out of the ec2 module_util at some point int he future but there is little reason why you should be manually writing each case.

This is the same for all the conversions you ahve done below.